docs/getting-started/understanding-config-files.md

```diff
@@ -453,43 +453,55 @@ At this point [Haddock
-      that it is permissible to share its address.
-
-The `PeerSharing` flag interacts with `PeerAdvertise` (`advertise` flag in the topology
-file) values as follows:
-
-`AdvertisePeer` (`advertise: true`) is local to the configuration of a specific node. A
-node might be willing to share those peers it has set as `PeerAdvertise`. Conversely,
-`PeerSharing` is about whether the peer (itself) is willing to participate in
-`PeerSharing`.
-
-`PeerSharing` takes precedence over `AdvertisePeer`. Consider the following example:
-
-A block-producing node (BP) has the `NoPeerSharing` flag value (which means it won't participate
-in peer sharing or run the mini-protocol). A relay node has the BP set as a local peer
-configured as `AdvertisePeer` (likely a misconfiguration). When the handshake between the
-BP and the relay occurs, the relay will see that the BP doesn't want to participate in
-peer sharing. As a result, it won't engage in peer sharing with it or share its details
-with others.
-
-In the handshake, if either party has `PeerSharingDisabled`, that value will be negotiated, resulting in the absence of any peer sharing protocol between the two parties. Peer sharing will only be enabled if both parties have `PeerSharingEnabled`.
+    * `True`: peer sharing is enabled, the node may issue as well as respond
+      to requests for peers.
+
+When `advertise: true` is set in the topology file for a given local root peers
+group, the node can disseminate knowledge of such peers with other participants
+of the network through the peer-sharing mechanism. The node only sends IP
+addresses and port numbers through peer sharing, no additional information is
+shared, in particular, the node doesn't send DNS names.
+
+Let's consider an example configuration:
+
+* a block-producing node (BP) has the `PeerSharing: False` set in its
+  configuration file;
+* a relay node should set `advertise: false` for the BP in its local peer
+  section of its topology file.
+
+**IMPORTANT** When the handshake between the BP and the relay occurs, the relay will see that
+the BP doesn't want to participate in peer sharing. As a result, neither side
+will engage in peer sharing on this connection.  If the BP had `PeerSharing:
+True` it could ask its relays for additional peers to connect to and up talking
+directly to untrusted peers
+
+**IMPORTANT** Please note that if the relay set `advertise: true` for the BP in its
+topology file, it could leak the knowledge about the BP's IP address to the
+outside word.
+
+For two nodes to share peers between themselves, both of them
+have to run with `PeerSharing: True`, if either of them sets it to `False`
+sharing peers will not be possible in any direction.
+
+**IMPORTANT** Note that if BP connects to some other relay which doesn't have
+a BP entry in its local root peers, the BP's IP address might be
+leaked by it to the outside word, since that relay will not know that BP's IP
+address should not be advertised.
```

Comment:
We should add a note that we recommend that the BP use a firewall to block all incoming traffic towards the cardano-node port apart from the SPO's own relays.

Reply:
One can actually block all traffic, BP will connect to the relays which will reuse the connection.